runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -2162,6 +2172,28 @@ handleServerMessage(JITServer::ClientStream *client, TR_J9VM *fe, JITServer::Mes
          client->write(response, TR::Compiler->cls.isClassRefValueType(comp, clazz, cpIndex));
          }
          break;
+      case MessageType::ClassEnv_enumerateFields:
+         {
+         auto recv = client->getRecvData<TR_OpaqueClassBlock *>();
+         const TR::TypeLayout* layout = comp->typeLayout(std::get<0>(recv));
+
+         std::vector<TR::TypeLayoutEntry> entries;
+         std::vector<std::string> fieldNames;
+         std::vector<std::string> typeSignatures;
+         entries.reserve(layout->count());
+         fieldNames.reserve(layout->count());
+         typeSignatures.reserve(layout->count());
+         for (int32_t idx = 0; idx < layout->count(); ++idx)
+            {
+            const TR::TypeLayoutEntry &entry = layout->entry(idx);
+            entries.push_back(entry);
+            // both strings are null-terminated so we don't need to know length
+            fieldNames.push_back(std::string(entry._fieldname));
```

Comment:
In `J9::ClassEnv::enumerateFields` when we retrieve the data from the client there is an assumption that the strings are null terminated. I was under the impression that the `string` constructor does not copy the null terminating character. Maybe some implementations do that so that they can optimize the `c_str()` function, but I don't think it's guaranteed.

Reply:
Apparently, starting with C++11 it is guaranteed to have a null terminator:
```
The returned array is null-terminated, that is, data() and c_str() perform the same function.
```

Reply:
Yeah, I used `c_str()` under the assumption that the result would have a null-terminator